tests/test_app.py

```diff
@@ -31,14 +32,39 @@ def auth_headers(workspace="workspace", user="user", expiry=None, scope="view"):
 
 def test_validate_invalid_token():
     url = "/workspace/workspace/current"
-    token = auth_token(url)
+    signer = itsdangerous.Signer("badsecret")
+    token = create_raw_token(
+        dict(
+            url=urljoin(client.base_url, "/workspace/workspace"),
+            user="user",
+            expiry=datetime.now(timezone.utc) - timedelta(hours=1),
+            scope="view",
+        ),
```

Comment:
I really don't like how `dict` constructors look!  I think it's because they don't look like the repr of a dict so I'm used to the literal syntax, and this form was slower than the literal for for a long time.  There's no take away from this, I just don't like them 😁